test/e2e/specs/site-editor/style-variations.spec.js

```diff
@@ -210,6 +225,106 @@ test.describe( 'Global styles variations', () => {
 			'rgb(255, 239, 11)'
 		);
 	} );
+
+	test( 'can create custom style variations', async ( {
+		admin,
+		page,
+		siteEditorStyleVariations,
+		siteEditor,
+	} ) => {
+		await admin.visitSiteEditor( {
+			postId: 'gutenberg-test-themes/style-variations//index',
+			postType: 'wp_template',
+		} );
+		await siteEditor.enterEditMode();
+
+		// Assert no custom styles yet.
+		await siteEditorStyleVariations.browseStyles();
+		await page.getByText( 'No custom styles yet.' ).click();
+
+		// Change background color
+		await page
+			.getByRole( 'button', { name: 'Navigate to the previous view' } )
+			.click();
+		await page.getByRole( 'button', { name: 'Colors styles' } ).click();
+		await page
+			.getByRole( 'button', { name: 'Colors background styles' } )
+			.click();
+		await page
+			.getByRole( 'button', { name: 'Color: Cyan bluish gray' } )
+			.click();
+
+		// Create new style
+		await page
+			.getByRole( 'button', { name: 'Navigate to the previous view' } )
+			.click();
+		await page
+			.getByRole( 'button', { name: 'Navigate to the previous view' } )
+			.click();
+		await page.getByRole( 'button', { name: 'Browse styles' } ).click();
+		await page
+			.locator(
+				'.components-card__body > .components-flex > .components-button'
```

Comment:
Is there a more accessible selector here we can use? Preferably `getByRole`? If not, then can we fix it in the source code to make it more accessible (if it's in the scope of this PR)?

Reply:
Will work on this

Reply:
the `Button` component _should_  normally have the `button` or `link` role, depending on whether an `href` prop is passed.